### DISCUSSION TITLE
Text Selection from file properties

### DISCUSSION THREAD

**Discussion by WebMechanic** [sumatrapdfreader/sumatrapdf#2437] [Q&A] [answered]
Hello,

the file property dialog contains useful information to copy from **for citation**, but all text fields are read only for no apparent reason. Virtually all file viewers I use allow to select and copy file information like the file path, topic, title etc. I see no reason why access to these properties is blocked in the UI.

Also: if the a file's theme or description (pdf) field has a lot of content, the dialog size just increases horizontally and can't be resized.

It'd be nice if the information of a file could be easily selected and copied to make citation as well as file path names more accessible.

Thanks

**Comment by GitHubRulesOK** [comment, marked as answer]
File properties can be copied using ctrl D (Display Details) and ctrl C (Copy to Clipboard)
in newer version 3.4 pre-release for fonts you can used ctrl D twice to include the font list (but it not really of much value)

**Comment by WebMechanic** [comment]
ok, so pressing `Ctrl+D` opens that dialog {1} - that I could figure out from the menu entry.
pressing `Ctrl+C` **while it's open** copies its content to the clipboard 🤯
How remarkably unpredictable and unintuitive 😂 
it's not listed in https://www.sumatrapdfreader.org/docs/Keyboard-shortcuts though.
but now I know.

Thank you!
---
{1} `Alt+Return` would be a more common shortcut on Windows to display file properties

**Comment by tactilis** [comment]
> ok, so pressing `Ctrl+D` opens that dialog {1} - that I could figure out from the menu entry. 
> 
> {1} `Alt+Return` would be a more common shortcut on Windows to display file properties

Adobe Acrobat also uses Ctrl+D for file properties:

![image](https://user-images.githubusercontent.com/4060092/160255202-dfa5c89a-18a0-4ff8-8723-82d6032bdd0d.png)

**Reply by GitHubRulesOK** [reply]
Thanks for screenshot
note  the difference is 
SumatraPDF shortcut :-) for Ctrl+D is Alt F + r (R in English, P in française etc.)
whereas Acrobat shortcut for Ctrl+D is Alt F + e

**Comment by WebMechanic** [comment]
Appreciate your effort, but I explicitly wrote *a more common shortcut **on Windows*** not for "other PDF applications" let alone any "Adobe product".
I added this remark after I checked the list of supported operating systems, and guess what? It's not even a list, just Windows. Therefore I believe suggesting "a _more common_ shortcut on Windows" is legit. Also "more common" does not mean "the default" or "the standard".
Not that linguistics matters these days.

So here's a little UI lesson from a seasoned UI developer and UX consultant ...

Adobe's choice of shortcuts is only relevant with _their own set of applications_ where they aim to be consistent with both generic and identical commands. They're doing quit well.
And it's slightly irrelevant for the subject matter, because unlike SumatraPDF, Adobe software is also _multi platform_ (for the most part) and the `Option` key (the equivalent of `Alt`ernate on MacOS) is more or less occupied by the OS itself.

Single `Option+whatever` shortcuts inside programs that (also) run on Macs aren't really a thing or very, very rare at best (i.e. Blender), but rather used to provide "options" for any existing (main) commands based off `Command`, `Control` + `Shift`. Adobe does this a lot and very thoughtful. I wished Affinity would copy that, too :)

The same no-single-alt+key shortcut "rule" is true for many other _multi platform_ apps such as web browsers.

So if you want to do the nitpicking: on **Windows** `Alt+Return/Enter` ...
- in Windows Explorer opens File, Folder, Drive properties
- on a Desktop icon opens Shortcut properties
- on "My Computer" gives you System Settings
- inside (older) M$ apps opened document properties (they changed this with the advent of "online office")
- TextPad editor: file properties

and there are other *native* (Windows only) applications that use this pattern.
"Unfortunately" most modern apps are multi-platform (or effing Electron based) so this key combo is now rare -- or there's no shortcut at all to quickly reach for file/document properties, which often is a pita.

> whereas Acrobat shortcut for Ctrl+D is Alt F + e

Because the letter `r` has already been assigned to "Convert to Wo**r**d, Excel or PowerPoint". 
Presumably a deliberate choice of Adobe, to be consistent among the many localizations of Acrobat: "Word" as product name remains "Word" in virtually any translation, whereas "properties" is a translatable UI string. The 2nd key (after pressing `Alt+F`) for this feature will or course change depending on the localization.

On **Windows**, however, `Alt+F`is simply based on the _first letter_ in "**F**ile" or "**F**ichier" which is a top level menu item. This keystroke would also vary in different locales, i.e. "**D**atei" (german), "**A**rchivo" (spanish), "**P**lik" (polish) as the Windows GUI guidelines for such menu commands recommend for consistency: use the first letter of a command's noun or action word as the 2nd level shortcut key.
Simple mnemonics or in other words: don't just make shit up. 😎

Have a nice day.👋🏻

**Reply by GitHubRulesOK** [reply]
see new "feature" of https://github.com/sumatrapdfreader/sumatrapdf/issues/2439
